clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -1961,9 +1967,8 @@ void CXXNameMangler::mangleLambda(const CXXRecordDecl *Lambda) {
   // mangling number for this lambda.
   llvm::Optional<unsigned> DeviceNumber =
       Context.getDiscriminatorOverride()(Context.getASTContext(), Lambda);
-  unsigned Number = DeviceNumber.hasValue() && *DeviceNumber
-                        ? *DeviceNumber
-                        : Lambda->getLambdaManglingNumber();
+  unsigned Number =
+      DeviceNumber ? *DeviceNumber : Lambda->getLambdaManglingNumber();
 
   assert(Number > 0 && "Lambda should be mangled as an unnamed class");
   if (Number > 1)
```

Comment:
What happens for 3 here?  Would it not conflict with 1?

Reply:
No, because '3' would mangle the number '1', '1' would mangle 'nothing'.

Reply:
Thanks, so there is {"unnamed class", nothing, 0,1,...} for {0, 1, 2, 3, ...} respectively.   Did I get that right?

Reply:
exactly.  Isn't Itanium fun :)?